src/app/config.cpp

```diff
@@ -150,12 +147,49 @@ Config *Config::get_instance()
 
 void Config::load_brightness_plugins()
 {
+    this->brightness_plugins["auto"] = QFileInfo();
+
     for (const QFileInfo &plugin : Config::plugin_dir("brightness").entryInfoList(QDir::Files)) {
         if (QLibrary::isLibrary(plugin.absoluteFilePath()))
             this->brightness_plugins[Config::fmt_plugin(plugin.baseName())] = plugin;
     }
 }
```

Comment:
Should `load_brightness_plugins` check `is_supported` and exclude any unsupported brightness plugins from the `brightness_plugins` map so that they can't be accidentally configured if we know they aren't going to work?

Reply:
so that would require each plugin to be loaded.... maybe use a temp `QPluginLoader` to do that?

Reply:
Yeah can probably do it that way.
I'm thinking I can handle the supported and priority check within `load_brightness_plugins` too and just store the name of the highest priority which is used in place of `auto`.

Reply:
so what you're saying is not having the "auto" mode shown, and just have the plugin auto-selected if the config doesn't contain anything?

Reply:
na, I already made the changes I was talking about so you can see it in the PR.

I think keeping the `auto` is good because it means its clear when the auto detection is being used.
(Maybe I should make it display what brightness plugin was selected on the settings page too?)

All I meant is say someone has a system that can't run `randr` for the `x` brightness control (for whatever reason, idk) we would exclude that from being shown on the settings page since we know we can't run that plugin.

Reply:
oh yeah i liked auto, just wanted to make sure we're on the same page hah